addon/mixins/select.js

```diff
@@ -52,43 +32,83 @@ export default Mixin.create({
     }
   },
 
+  label: computed('value', 'content', function () {
+    let checkValue = this.get('value')
+
+    if (typeof this.get('value') === 'string' || typeof this.get('value') === 'boolean' ) {
+      checkValue = this.findValueObject(this.get('value'));
+    }
+
+    if (checkValue === null || checkValue === undefined) {
+      return this.get('prompt')
+    }
```

Comment:
In my `gf_expire` story, one of the options for "this remains active for..." is "Forever"... which is `null`. I think it should be okay for `null` to be a valid option value. Whenever there's a nullable database field, the potential for nullity should be present everywhere.

Maybe `if (checkValue === null) { return this.findValueObject(checkValue) || this.get('prompt') }`? That way, if an option _does_ have `value: null`, we use it.

Reply:
ok.  i am a little confused right now so please forgive me for any confusion but i think right now we can have `this.get('value')` as `null`.  so are you wanting the label to be showing as `null` as well instead of the prompt?

Reply:
ok.  talked to @lawrencegetfeedback  and understanding what the issue is.  i looked at the `gf_expire` branch and it seems like it still works for the user right now.  the experience might look a little weird for the user right now since `Forever` is showing up as the prompt but it seems to be saving.  we are working on a bigger refactor of the `bourbon-search-select-field` and `bourbon-select-field` that would fix this but it would be nice to get this one out right now that fixes the other issues for the users.  does that work for you @amacdougall ?

Reply:
you can see here https://github.com/MatchbookLabs/bourbon/compare/master...lawrence-bourbon-select-field-refactor-base . @lawrencegetfeedback has made a PR to simplify the component by converting everything into an Ember Object and we can just treat everything the same.  And not worry about the format of the data coming in.

Reply:
Works for me! Sorry for the confusion (and for not seeing this a bit earlier). As far as I'm concerned, we can make the prompt also say "Forever", and then the user won't notice a thing.